### DISCUSSION TITLE
useTokenApprove.  useTokenPermit

### DISCUSSION THREAD

**Discussion by lxhyl** [wevm/wagmi#867] [Ideas]
These are frequently used method. Can we support?

I would like to do this job if possible.

**Comment by jxom** [comment]
Thanks for the idea! `wagmi` aims to keep it's API surface very low, and our API is typically one-to-one with [Ethers](https://docs.ethers.io/v5/). That said, we can definitely encourage you to write a [Custom Hook](https://reactjs.org/docs/hooks-custom.html) for this and open a PR to [`awesome-wagmi`](https://github.com/wagmi-dev/awesome-wagmi).
